packages/strapi-plugin-users-permissions/services/User.js

```diff
@@ -0,0 +1,127 @@
+'use strict';
+
+/**
+ * User.js service
+ *
+ * @description: A set of functions similar to controller's actions to avoid code duplication.
+ */
+
+// Public dependencies.
+const _ = require('lodash');
+const bcrypt = require('bcryptjs');
+
+module.exports = {
+
+  /**
+   * Promise to fetch all users.
+   *
+   * @return {Promise}
+   */
+
+  fetchAll: (params) => {
+    const convertedParams = strapi.utils.models.convertParams('user', params);
+
+    return strapi.plugins['users-permissions'].models.user
+      .find()
+      .where(convertedParams.where)
+      .sort(convertedParams.sort)
+      .skip(convertedParams.start)
+      .limit(convertedParams.limit)
+      .populate(_.keys(_.groupBy(_.reject(strapi.plugins['users-permissions'].models.user.associations, {autoPopulate: false}), 'alias')).join(' '));
```

Comment:
To harmonise with the queries used above and be more understandable, maybe it's better to write this?
 
```js
.populate(
  strapi.plugins['users-permissions'].models.user.associations
  .filter(x => x.autoPopulate)
  .map(x => x.alias)
  .join(' ')
)
```